### PR TITLE
feat: show legends per default

### PR DESCRIPTION
**Pull request by TreffN**
This PR adds an option to set the layer legends visible by default when the layer is visible, so you don't need to select the legend in the context menu. 
Settings in the tool-config in SHOGun-Admin:
{
    "name": "tree",
    "config": {
      "showLegends": false,
      "visible": true
    }
  },
Legenden1

changing the showLegends config to true gives this initial view in the SHOGun-Gis-Client:
Legenden2

Please review :)

### CODE REVIEW

File: src/bootstrap.tsx [hunk 1]
```diff
@@ -91,7 +91,10 @@ import {
   EditLevel
 } from './store/editFeature';
 import { setFeatureInfoActiveCopyTools } from './store/featureInfo';
-import { setLayerTreeActiveUploadTools } from './store/layerTree';
+import {
```
Comment (faouziH21) [inline src/bootstrap.tsx:94]:
``` suggestion 
import {
```

File: src/components/ToolMenu/LayerTree/index.tsx [hunk 2]
```diff
@@ -72,11 +74,15 @@ export const LayerTree: React.FC<LayerTreeProps> = ({
     t
   } = useTranslation();
 
-  const [visibleLegendsIds, setVisibleLegendsIds] = useState<string[]>([]);
-  const [layerTileLoadCounter, setLayerTileLoadCounter] = useState<LayerTileLoadCounter>({});
+  const showLegendsState = useAppSelector(state => state.layerTree.showLegends);
+  //check if showLegends is defined
```
Comment (faouziH21) [inline src/components/ToolMenu/LayerTree/index.tsx:78]:
```suggestion 
// check if showLegends is defined
```

**Comment by faouziH21** [comment]
Nice! I Have one question: Does this cause a lot of scrolling in the case that there are a lot of layers? If so maybe we should consider making the legend collapsible. But all in all well done :D

File: src/bootstrap.tsx (hunk 1 above)
Comment (TreffN) [reply src/bootstrap.tsx]: fixed in new commit

File: src/components/ToolMenu/LayerTree/index.tsx (hunk 2 above)
Comment (TreffN) [reply src/components/ToolMenu/LayerTree/index.tsx]: fixed in newest commit

**Review by LukasLohoff — APPROVED** [review]

File: src/components/ToolMenu/LayerTree/index.tsx [hunk 3]
```diff
@@ -61,6 +62,7 @@ export type LayerTileLoadCounter = {
 export type LayerTreeConfig = {
   enabled?: boolean;
   activeUploadTools?: UploadTools[];
+  showLegends?: boolean;
 };
```
Comment (dnlkoch) [inline src/components/ToolMenu/LayerTree/index.tsx:66]: Not directly related to the changes in this PR, but it looks like the type `LayerTreeConfig` is used in the store only. Probably we should think about moving it over.

File: src/components/ToolMenu/LayerTree/index.tsx [hunk 4]
```diff
@@ -72,11 +74,15 @@ export const LayerTree: React.FC<LayerTreeProps> = ({
     t
   } = useTranslation();
 
-  const [visibleLegendsIds, setVisibleLegendsIds] = useState<string[]>([]);
-  const [layerTileLoadCounter, setLayerTileLoadCounter] = useState<LayerTileLoadCounter>({});
+  const showLegendsState = useAppSelector(state => state.layerTree.showLegends);
+  // check if showLegends is defined
+  const showLegends: boolean = showLegendsState ? showLegendsState : false;
```
Comment (dnlkoch) [inline src/components/ToolMenu/LayerTree/index.tsx:79]: Is this really needed? Can't we use `showLegendsState` directly?

File: src/components/ToolMenu/LayerTree/index.tsx [hunk 5]
```diff
@@ -72,11 +74,15 @@ export const LayerTree: React.FC<LayerTreeProps> = ({
...
 
   const initialLayersUid = map?.getAllLayers().map(l => getUid(l));
 
+  const [visibleLegendsIds, setVisibleLegendsIds] = useState<string[]> (showLegends? initialLayersUid ?? [] : []);
```
Comment (dnlkoch) [inline src/components/ToolMenu/LayerTree/index.tsx:83]:
```suggestion
  const [visibleLegendsIds, setVisibleLegendsIds] = useState<string[]> (showLegends ? initialLayersUid ?? [] : []);
```

**Review by dnlkoch — APPROVED** [review]

File: src/components/ToolMenu/LayerTree/index.tsx (hunk 3 above)
Comment (TreffN) [reply src/components/ToolMenu/LayerTree/index.tsx]: Moved the type LayerTreeConfig to the store

File: src/components/ToolMenu/LayerTree/index.tsx (hunk 4 above)
Comment (TreffN) [reply src/components/ToolMenu/LayerTree/index.tsx]: Only use of showLegendsState in the new commit

File: src/components/ToolMenu/LayerTree/index.tsx (hunk 5 above)
Comment (TreffN) [reply src/components/ToolMenu/LayerTree/index.tsx]: see new commit

**Review by LukasLohoff — APPROVED** [review]

**Pull request merged**